CMakeLists.txt

```diff
@@ -172,8 +172,21 @@ set(PICOHTTP_TEST_LIBRARY_FILES
     picoquictest/h3zerotest.c
 )
 
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake")
-find_package(PTLS REQUIRED)
+option(ENABLE_FETCH_CONTENT "use cmake fetch-content for dependency management" OFF)
+
+if(ENABLE_FETCH_CONTENT)
+  include(FetchContent)
+  FetchContent_Declare(   picotls
+      GIT_REPOSITORY      https://github.com/h2o/picotls.git
+      GIT_TAG             047c5fe20bb9ea91c1caded8977134f19681ec76)
+  FetchContent_MakeAvailable(picotls)
+  set(PTLS_INCLUDE_DIRS ${picotls_SOURCE_DIR}/include)
+  set(PTLS_LIBRARIES picotls-core picotls-openssl picotls-fusion)
+else()
```

Comment:
The "set PTLS_LIBRARIES" above is incorrect, and does not produce the same results as "find_package(PTLS REQUIRED)". The "find" call will execute (picoquic)/cmake/findPTLS.cmake, which checks whether picotls was compiled with or without the fusion libraries and sets the variables PTLS_INCLUDE_DIRS and PTLS_LIBRARIES accordingly. That code must be executed after "make picotls".

I am not sure what FetchContent_MakeAvailable does. If it triggers a "make" of picotls, it might be sufficient to replace the two "set" instructions above by "find_package(PTLS REQUIRED)".

Reply:
Also, I would add a couple of comment lines (message(STATUS "..." ) to track what is going on. The output of the github action includes:
```
--   Enabling OpenSSL support
-- picotls/include: /Users/runner/work/picoquic/picoquic/_deps/picotls-src/include
-- picotls libraries: picotls-core;picotls-openssl;picotls-fusion
-- PTLS compiled without support for Fusion
```
This is an error -- either we have support for fusion or we do not include "picotls-fusion" in the library lists.

Reply:
@huitema  I now understand the reason for failure, but don't have a good answer on how to resolve it.

Cause: picoquic cmake sets option ``` PTLS_WITH_FUSION ``` based on if the libpicotls-fusion.a was found or not. This enforces the user of picoquic to have picotls built before hand and findPTLS will end up finding it. 

However with fetch_content , the picotls is fetched on cmake run and builds as part of building picoquic. Hence it is unable to preset the above option ``` PTLS_WITH_FUSION ``` before linking PTLS.

Proposal: Would it be acceptable if I set ``` PTLS_WITH_FUSION ``` for osx builds to start with when using fetch_content as below
``` cmake -DENABLE_FETCH_CONTNET=ON -DPTLS_WITH_FUSION=ON ``` .. 

This should make the builders happy but the burden is for us to choose when to set the above option and on what platforms

Reply:
I don't like hard coded options. It is OK for experimentation, but not a final solution. Opened an issue in picotls repo: https://github.com/h2o/picotls/issues/392

Reply:
I think I understand now what is going on. The failure in the GitHub action is not because of fusion but because of "brotli". The PTLS CMakeList has code to detect brotli:
```
PKG_CHECK_MODULES(BROTLI_DEC libbrotlidec)
PKG_CHECK_MODULES(BROTLI_ENC libbrotlienc)
IF (BROTLI_DEC_FOUND AND BROTLI_ENC_FOUND)
    INCLUDE_DIRECTORIES(${BROTLI_DEC_INCLUDE_DIRS} ${BROTLI_ENC_INCLUDE_DIRS})
    LINK_DIRECTORIES(${BROTLI_DEC_LIBRARY_DIRS} ${BROTLI_ENC_LIBRARY_DIRS})
    SET(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DPICOTLS_USE_BROTLI=1")
    LIST(APPEND CORE_FILES
        lib/certificate_compression.c)
    LIST(APPEND CORE_EXTRA_LIBS ${BROTLI_DEC_LIBRARIES} ${BROTLI_ENC_LIBRARIES})
ENDIF ()
```
In the successful runs on Ubuntu, the test fails:
```
-- Checking for module 'libbrotlidec'
--   No package 'libbrotlidec' found
-- Checking for module 'libbrotlienc'
--   No package 'libbrotlienc' found
```
That means the code on Ubuntu has no reference to "brotli". But on MAC/OS the test succeeds:
```
-- Checking for module 'libbrotlidec'
--   Found libbrotlidec, version 1.0.9
-- Checking for module 'libbrotlienc'
--   Found libbrotlienc, version 1.0.9
```
But then, there is no directive to find that component in the picoquic build, and the library is not found.